src/options-resolver.ts

```diff
@@ -15,33 +15,89 @@
  * limitations under the License.
  *
  */
+import chalk from 'chalk'
+import * as prompts from 'prompts'
+import {Choices, ConfirmOptionType, InputOptionType, SelectOptionType} from './interfaces/options.interface'
+import {Validator} from './validators/validator'
 
-import * as readlineSync from 'readline-sync'
-
-export const OptionsResolver = (options: any,
-                                key: string,
-                                secondSource: () => string | undefined,
-                                promptText: string,
-                                readlineDependency?: any,
-                                hide?: boolean): any => {
-    const readline = readlineDependency || readlineSync
-    const hideEchoBack = hide ? true : false
-    return options[key] !== undefined ? options[key] : (secondSource() ||
-        readline.question(promptText, {hideEchoBack}))
+export const OptionsChoiceResolver = async (options: any,
+                                            key: string,
+                                            promptText: string,
+                                            choices: Choices[],
+                                            type: SelectOptionType = 'select',
+                                            validation: Validator<any> | undefined) => {
+    if (!['select', 'multiselect'].includes(type)) {
+        console.log(chalk.red('ERR'), 'Invalid options choice resolver type')
+        return process.exit()
+    }
```

Comment:
Maybe we could remove these codes. Because I defined a customized type `SelectOptionType` to limit this value.
[src/interfaces/options.interface.ts](https://github.com/nemtech/symbol-cli/blob/56be4409bcc67d16278659e811871ef216f27bdb/src/interfaces/options.interface.ts)

Reply:
Solved 👍